src/base/asyncfilestorage.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2017  Vladimir Golovnev <glassez@yandex.ru>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "asyncfilestorage.h"
+
+#include <QDebug>
+#include <QMetaObject>
+#include <QMutexLocker>
+#include <QSaveFile>
+#include <QTimer>
+
+AsyncFileStorage::AsyncFileStorage(const QDir &storageDir, QFile *lockFile, QObject *parent)
+    : QObject(parent)
+    , m_storageDir(storageDir)
+    , m_lockFile(lockFile)
+{
+}
+
+AsyncFileStorage::~AsyncFileStorage()
+{
+    m_lockFile->close();
+    m_lockFile->remove();
+    delete m_lockFile;
+}
+
+AsyncFileStorage *AsyncFileStorage::create(const QString &storageFolder, QObject *parent)
+{
+    QDir storageDir(storageFolder);
+    if (storageDir.exists() || storageDir.mkpath(storageDir.absolutePath())) {
+        QFile *lockFile = new QFile(storageDir.absoluteFilePath(QStringLiteral("storage.lock")));
+        if (lockFile->open(QFile::WriteOnly))
+            return new AsyncFileStorage(storageDir, lockFile, parent);
+        delete lockFile;
+    }
+
+    return nullptr;
+}
+
+void AsyncFileStorage::store(const QString &fileName, const QByteArray &data)
+{
+    QMutexLocker locker(&m_jobsMutex);
+    m_jobs.insert(fileName, data);
+    qDebug() << "AsyncFileStorage: Added job for" << fileName;
+    if (m_isIdle) {
+        m_isIdle = false;
+        // The function will be run in the thread of receiver (this AsyncFileStorage object).
+        QTimer::singleShot(0, this, &AsyncFileStorage::processPendingJobs);
```

Comment:
>I want to simplify AsyncDataStorage by removing jobs rewriting. I want optimize the RSS classes to perform delayed data saving

So this "concurrency" code will be unneeded. 
Think of it as a temporary solution. I'll change it, perhaps even along with this PR. Either directly after it. As I wrote, I don't want to delay it, because the basic functionality is ready and it is required for other tasks that need to be on top of it (I mean so long-awaited web interface).